src/js/view/Input.js

```diff
@@ -1,15 +1,63 @@
 import { readLineAsync } from "../../utils/readlineAsync";
+import LottoGame from "../domain/LottoGame";
+import LottoNumber from "../domain/LottoNumber";
+
 const Input = {
   async getLottoPurchasedAmount() {
-    return await readLineAsync("구입금액을 입력해 주세요.");
+    while (true) {
+      try {
+        const purchasedAmount = await readLineAsync(
+          "구입금액을 입력해 주세요."
+        );
+        LottoGame.validateLottoPurchasedAmount(purchasedAmount);
+        return purchasedAmount;
+      } catch (e) {
+        console.log(e.message);
+      }
+    }
```

Comment:
```suggestion
    try {
      const purchasedAmount = await readLineAsync(
        "구입금액을 입력해 주세요."
      );
      LottoGame.validateLottoPurchasedAmount(purchasedAmount);
      return purchasedAmount;
    } catch (e) {
      console.log(e.message);
      Input.getLottoPurchasedAmount();
    }
```
이렇게 표현할 수도 있을 것 같네요!
아니면 에러가 그냥 전파되도록 두고, main 함수에서 catch해서 다시 input을 호출하는 방법도 있을 것 같아요!

그렇게 되면 input의 입장에서는 말 그대로 입력값만 받아오고,
검증은 domain에게 넘기고
retry는 main에서 하는거죠 ㅎㅎ

그럼 역할과 책임이 딱딱 맞아떨어지지 않나요!?

Reply:
그렇네요! 반복 로직은 Input 도메인에서 제거하였습니다. (`IO.test.js` 에서도 retry 관련 테스트들을 제거하였습니다.)

또한, 다음과 같이 retry 하는 로직은 util 로 만들어 main 에서 사용하도록 해보았습니다. 

```javascript
const repeatUntilNoError = async (callback) => {
  while (true) {
    try {
      const input = await callback();
      return input;
    } catch (e) {
      console.error(e.message);
    }
  }
};

export default repeatUntilNoError;

```